### PR TITLE
Docs for command_track

### PR DESCRIPTION
**Pull request by thekafkaf**
Very simple code, no need for much documentation, documented mostly the functions(godoc is awesome).

### CODE REVIEW

File: commands/command_track.go [hunk 1]
```diff
@@ -69,11 +71,15 @@ func trackCommand(cmd *cobra.Command, args []string) {
...
 type mediaPath struct {
-	Path   string
-	Source string
+	Path   string // the tracked path
+	Source string // the source attributes file
 }
 
+// findAttributeFiles returns the(existing) paths of the attributes files
+// (./.git/info/attributes, ./.gitattributes).
+// Which is where the tracked paths are configured.
```
Comment (technoweenie) [inline commands/command_track.go]: How about something like this:

> findAttributeFiles returns the absolute paths to files in the current repository that Git uses to track attributes.

File: commands/command_track.go [hunk 2]
```diff
@@ -96,6 +102,8 @@ func findAttributeFiles() []string {
 	return paths
 }
 
+// findPaths parsers the existing attributes files, and extracts the tracked paths
+// as an array of mediaPaths.
```
Comment (technoweenie) [inline commands/command_track.go]: > findPaths extracts the paths tracked by Git LFS from the existing Git attributes files.

File: commands/command_track.go [hunk 3]
```diff
@@ -19,6 +19,8 @@ var (
 	}
 )
 
+// trackCommand takes a list of paths as an argument, and adds each path to the default
+// attribtues file (.gitattributes), if it's not already exist.
```
Comment (synapsos) [inline commands/command_track.go]: This should be:
``// attributes file (.gitattributes), if it not already exists.``

File: commands/command_track.go (hunk 2 above)
Comment (thekafkaf) [inline commands/command_track.go]: Used to python, so automatically documented the return type :P

**Comment by bkeepers** [comment]
@thekafkaf thanks for the tweaks.

@technoweenie @rubyist look ok to you?

**Comment by technoweenie** [comment]
Whoop, I missed the extra commits.  There's a few slight grammar changes I'd like to make first.

File: commands/command_track.go [hunk 4]
```diff
@@ -19,6 +19,8 @@ var (
 	}
 )
 
+// trackCommand takes a list of paths as an argument, and adds each path to the default
+// attribtues file (.gitattributes), if it's not already exists.
```
Comment (michael-k) [inline commands/command_track.go]: attribtues → attributes

Also, not completely true. It sounds like there is only one `.gitattributes` file.  Actually there is one per directory.

File: commands/command_track.go [hunk 5]
```diff
@@ -124,13 +130,18 @@ func findPaths() []mediaPath {
 	return paths
 }
 
+// needsTrailingLinebreak returns 'true' if a file doesn't end in a newline,
+// 'false' otherwise (even if an error occures).
```
Comment (ttaylorr) [inline commands/command_track.go]: I think this may be a misspelling of "occurs".

File: commands/command_track.go [hunk 6]
```diff
@@ -124,13 +130,18 @@ func findPaths() []mediaPath {
...
 func needsTrailingLinebreak(filename string) bool {
 	file, err := os.Open(filename)
 	if err != nil {
 		return false
 	}
 
 	defer file.Close()
+
+	// Reading the file in chuncks of 16384 bytes,
```
Comment (ttaylorr) [inline commands/command_track.go]: "chuncks" may be a misspelling of "chunks".

**Comment by ttaylorr** [comment]
Thanks for this, @thekafkaf! Pending the few minor misspellings I noted, along with @michael-k's comment about the `trackCommand` documentation, this should be good to go 👍

**Comment by ttaylorr** [comment]
Going to close this one out for now, if you want to come back to this in the future, please let me know! 😄

**Pull request closed without merging**